### PR TITLE
Verify ./runner executable on all supported OSes

### LINKED ISSUE #10126: Run whole Standard.Base in Native Image ./runner
Now, when the PR
- #9866

is integrated we can run parts of `test/Base_Tests` in the CI as `./runner --run test/Base_Tests ^Text`. The goal of this issue is to investigate what is wrong and remove the `^Text` argument - e.g. run all of the `test/Base_Tests` in _native image mode_ and thus make sure the whole `Standard.Base` library can be executed in such mode.

### PR DESCRIPTION
**Pull request by JaroslavTulach**
### Pull Request Description

Follow up to #10126. If we want to use the `./runner` executable as the default launcher, we need to make sure it builds and runs on all operating systems we support.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] All code follows the style guides
- [x] All CI checks continue to pass

### CODE REVIEW

**Comment by JaroslavTulach** [comment]
I guess all the [Base_Tests](https://github.com/enso-org/enso/actions/runs/9609440170/job/26503931081?pr=10329#step:7:3522) continue to run without any error.

File: engine/runner/src/main/resources/META-INF/native-image/org/enso/runner/reflect-config.json [hunk 1]
```diff
@@ -1800,458 +1800,6 @@
   "name":"org.enso.base_test_helpers.RangeStream",
   "methods":[{"name":"<init>","parameterTypes":["int", "int"] }]
 },
-{
```
Comment (JaroslavTulach) [inline engine/runner/src/main/resources/META-INF/native-image/org/enso/runner/reflect-config.json:1803]: Additional NI related fix:

`Persistance` discovers all the available implementations via `META-INF/services/` lookup. That would require reflective access to constructor. However such a discovery is now only performed in _static initializer of the class_ - e.g. during _build time_ of Native image. Hence no reflective operations with `Persistance` subclasses shall be performed during native executable execution. Removing _persistance related debris_ from the reflection configuration file.

**Comment by JaroslavTulach** [comment]
- We seem to have **space in path** issue on Mac: https://github.com/enso-org/enso/actions/runs/9610005652/job/26505581593?pr=10328#step:7:1546

- we cannot execute `smalljdk.sh`!? on windows: https://github.com/enso-org/enso/actions/runs/9610005652/job/26505582261?pr=10328#step:7:949 - why do we have the shall script at all?

Both issues shall be solved by 71b20a7

**Comment by JaroslavTulach** [comment]
The [windows job](https://github.com/enso-org/enso/actions/runs/9611701550/job/26510820796?pr=10329#step:7:1688) has succeeded in **14min**. The [Mac OS X job](https://github.com/enso-org/enso/actions/runs/9611701550/job/26510820217?pr=10329#step:7:2122) succeeded as well but it took one hour. Obviously [Linux job](https://github.com/enso-org/enso/actions/runs/9611701550/job/26510819862?pr=10329#step:7:3665) was successful in **23min**. We could make these numbers lower by removing the NI+Espresso build - that one is highly experimental and needs only to be executed on a single platform.

**Update** after 1b0d22e that removed Espresso+NI check on Mac and Windows we get :
- Linux runs for **23min** (still with Espresso+NI check)
- Mac finished in **46min**
- Windows tool **12min**

File: build.sbt [hunk 2]
```diff
@@ -2579,12 +2579,19 @@ lazy val `engine-runner` = project
...
-
+      var jlinkArgs = Seq(
+        "--module-path",
+        modules.mkString(File.pathSeparator),
+        "--output",
+        smallJdkDirectory.toString(),
+        "--add-modules",
+        s"$NI_MODULES,$JDK_MODULES,$DEBUG_MODULES,$PYTHON_MODULES"
+      )
+      val exitCode = scala.sys.process.Process(jlink.toString(), jlinkArgs).!
```
Comment (JaroslavTulach) [inline build.sbt:2590]: @jdunkerley reports https://github.com/enso-org/enso/actions/runs/9184711886/job/26530345747 failure on Mac Arm. It looks like a **space in path** issue and it should be fixed by this change.

File: build/cli/src/lib.rs [hunk 3]
```diff
@@ -433,9 +433,9 @@ impl Processor {
             arg::backend::Command::CiCheck {} => {
                 let config = enso_build::engine::BuildConfigurationFlags {
                     build_benchmarks: true,
-                    // Windows is not yet supported for the native runner.
-                    build_native_runner: enso_build::ci::big_memory_machine()
```
Comment (JaroslavTulach) [inline build/cli/src/lib.rs:437]: I am wondering why we `build_native_runner` on [Mac Arm](https://github.com/enso-org/enso/pull/10329/files#r1649552868)? Probably the Mac Arm classifies as `big_memory_machine`...

**Review by jdunkerley — APPROVED** [review]

**Review by 4e6 — APPROVED** [review]

**Review by Akirathan — APPROVED** [review]

**Review by hubertp — APPROVED** [review]

**Pull request merged**